src/pykeen/pipeline/api.py

```diff
@@ -915,6 +915,15 @@ def pipeline(  # noqa: C901
     if evaluator_kwargs is None:
         evaluator_kwargs = {}
     evaluator_kwargs = dict(evaluator_kwargs)
+    if evaluator_kwargs.get('filtered'):
+        if use_testing_data and validation is not None:
```

Comment:
So what should happen in the scenario when I do
```
from pykeen.pipeline import pipeline

results = pipeline(model=..., dataset=...)
```

It will also include the validation set (which wasn't used for anything) in the evaluation to remove false negatives? Is this fair, since it basically tells the evaluator extra information about what the model will probably get wrong, even though the model didn't train on it?

How would this be different from combining the testing + validation set in this scenario?

I get that this makes sense in the HPO scenario, though

Reply:
> So what should happen in the scenario when I do
> 
> ```
> from pykeen.pipeline import pipeline
> 
> results = pipeline(model=..., dataset=...)
> ```
> 
> It will also include the validation set (which wasn't used for anything) in the evaluation to remove false negatives? Is this fair, since it basically tells the evaluator extra information about what the model will probably get wrong, even though the model didn't train on it?
> 
> How would this be different from combining the testing + validation set in this scenario?
> 
> I get that this makes sense in the HPO scenario, though

The idea is that when using one of the pre-defined workflows, we stick to the "standard" training/evaluation procedure. For evaluating a model on the test set, this would mean that we add the validation triples to the set of known triples as proposed by (Bordes et al., 2013), even though one didn't make use of the validation triples before.
However, when using an evaluator in custom workflows, users can flexibly extend the set of known triples.

Reply:
We have to configure this for the paper based configurations that are in place, because in a normal setting the authors performed a model pre-selection by finding the best parameters using the validation set. Therefore, it is only reasonable to include the validation set during the final test evaluation as proposed by Bordes et al., 2013, since that is also what they did in their paper. However, Charly is right that when a user is training a dataset and never uses the validation set it correctly should not be included during the final test evaluation either.

Reply:
@lvermue you introduced the `filter_validation_when_testing` with a default value of true for the pipeline. Shouldn't the default value be false? But when doing HPO, explicitly set to true?

Reply:
@cthoyt  No, since the pipeline usually uses an early stopper that also uses the validation triples.
The main reason to default to true is that as explained in Bordes et al., 2013 models are usually optimized on the validation dataset. Therefore, when evaluation on the test dataset the validation triples have still to be filtered.
Since most users will use a model from a publication, there are two good reasons to use it as default. First of all, the model's parameters from the publication were derived from the validation dataset, thus it should also be filtered. 
Second, because of that the publications report their results usually filtering the validation dataset in the final test evaluation too, the results are only comparable when doing so too even when the user is training new configurations without touching the validation triples.

Reply:
Okay, thanks for the explanation. I appreciate the extra thought - I think all of my concerns are addressed

Reply:
I updated the pipeline() docs with a shorter version of this in21e413db